Doc/library/socket.rst

```diff
@@ -1485,28 +1485,36 @@ to sockets.
    .. availability:: not WASI.
 
 
-.. method:: socket.close()
+.. method:: socket.close(shutdown=False)
 
-   Mark the socket closed.  The underlying system resource (e.g. a file
-   descriptor) is also closed when all file objects from :meth:`makefile`
-   are closed.  Once that happens, all future operations on the socket
-   object will fail. The remote end will receive no more data (after
-   queued data is flushed).
+   Mark the socket closed. The underlying system resource (e.g., a file descriptor) is also
+   closed when all file objects from :meth:`makefile` are closed. Once that happens, all future
+   operations on the socket object will fail. The remote end will receive no more data
+   (after queued data is flushed).
+
+   If the ``shutdown`` parameter is set to ``True``, the socket will first be
+   shut down before closing, ensuring that no further data can be sent or received.
+   This is useful for properly releasing resources and preventing issues like lingering
```

Comment:
- Just a small suggestion `persistent connection` would be more accurate term, instead of `lingering connection`

```suggestion
   This is useful for properly releasing resources and preventing issues like persistent
```